### PR TITLE
refact(sysfs): add a generic sysfs package for handling sysfs on the host

### PR DESCRIPTION
**Pull request by akhilerm**
**Why is this PR required? What issue does it fix?**:
This is a refactoring PR that consolidates changes from multiple packages that use the sysfs to a single package.

**What this PR does?**:
All methods for accessing and parsing sysfs are moved to the sysfs package. The hierarchy package has been merged
and functionality from sysfs probe has been moved out.

**Does this PR require any upgrade changes?**:
No

**If the changes in this PR are manually verified, list down the scenarios covered:**:

**Any additional information for your reviewer?** : 
_Mention if this PR is part of any design or a continuation of previous PRs_

**Checklist:**
- [ ] Fixes #<issue number>
- [x] PR Title follows the convention of  `<type>(<scope>): <subject>`
- [ ] Has the change log section been updated? 
- [x] Commit has unit tests
- [ ] Commit has integration tests
- [ ] (Optional) Are upgrade changes included in this PR? If not, mention the issue/PR to track: 
- [ ] (Optional) If documentation changes are required, which issue on https://github.com/openebs/openebs-docs is used to track them:

### CODE REVIEW

File: pkg/sysfs/syspath.go [hunk 1]
```diff
@@ -0,0 +1,318 @@
...
+	// It is kept as 512 bytes. all entries in /sys/class/block/sda/size
+	// are in 512 byte blocks
+	sectorSize int64 = 512
+)
+
+var sysFSDirectoryPath = "/sys/"
+
+// getDeviceSysPath gets the syspath struct for the given blockdevice.
+// It is generated by evaluating the symlink in /sys/class/block.
+func getDeviceSysPath(devicePath string) (string, error) {
```
Comment (harshthakur9030) [inline pkg/sysfs/syspath.go:43]: @akhilerm how will NVMe devices be handled by this function?

Comment (akhilerm) [reply pkg/sysfs/syspath.go:43]: Yes @harshthakur9030 . All block devices will have an entry in `/sys/class/block/`. Only the symlink it points to will be different.

File: pkg/sysfs/syspath.go [hunk 2]
```diff
@@ -0,0 +1,320 @@
...
+	}
+	return numberOfBlocks * sectorSize, nil
+
+}
+
+// GetDeviceType gets the device type, as shown in lsblk
+// devicetype should be prefilled by udev probe (DEVTYPE) as disk/part for this to work
+//
+// Ported from https://github.com/karelzak/util-linux/blob/master/misc-utils/lsblk.c
+func (s Device) GetDeviceType(devType string) (string, error) {
```
Comment (akhilerm) [inline pkg/sysfs/syspath.go:268]: @richardelling  Can you review this piece of code. Its actually a port of `lsblk` from util-linux.

File: pkg/sysfs/device.go [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+
+	dev := &Device{
+		deviceName: devName,
+		path:       devPath,
+		sysPath:    sysPath,
+	}
+	return dev, nil
+}
+
+/*func NewSysFsDeviceFromSysPath(sysPath string) (*Device, error) {
```
Comment (kmova) [inline pkg/sysfs/device.go:52]: n_: remove this unused code

**Review by kmova — DISMISSED** [review]

File: pkg/sysfs/device.go (hunk 3 above)
Comment (akhilerm) [reply pkg/sysfs/device.go]: Done.

**Review by kmova — APPROVED** [review]

**Pull request merged**